projects/packages/sync/src/modules/class-module.php

```diff
@@ -340,19 +340,18 @@ public function send_full_sync_actions( $config, $status, $send_until ) {
 		$chunks_sent = 0;
 		// phpcs:ignore Generic.CodeAnalysis.AssignmentInCondition.FoundInWhileCondition
 		while ( $objects = $this->get_next_chunk( $config, $status, $limits['chunk_size'] ) ) {
-			if ( $chunks_sent++ === $limits['max_chunks'] || microtime( true ) >= $send_until ) {
-				return $status;
-			}
 
 			$result = $this->send_action( 'jetpack_full_sync_' . $this->name(), array( $objects, $status['last_sent'] ) );
-
 			if ( is_wp_error( $result ) || $wpdb->last_error ) {
 				$status['error'] = true;
 				return $status;
 			}
 			// The $ids are ordered in descending order.
-			$status['last_sent'] = end( $objects );
-			$status['sent']     += count( $objects );
+			$status = $this->set_send_full_sync_actions_status( $status, $objects );
+
+			if ( $chunks_sent++ === $limits['max_chunks'] || microtime( true ) >= $send_until ) {
```

Comment:
```suggestion
			if ( ++$chunks_sent === $limits['max_chunks'] || microtime( true ) >= $send_until ) {
```
So that we increment first, compare after :) 

Moving this check in the end of the loop is a performance optimization that will save us 1 DB query plus the overkill of expanding items before sending.